test/test_background_flux_divergence.jl

```diff
@@ -0,0 +1,78 @@
+using Oceananigans
+using Test
+using CUDA:has_cuda
+using NCDatasets
+
+"""
+run_with_background_fields(arch; with_background=true)
+Run a model with or without background fields and compare the two.
+"""
+function run_with_background_fields(arch; with_background=true)
+    grid = RectilinearGrid(arch, size=10, z=(0, 1), topology=(Flat, Flat, Bounded))
+    # Setup model with or without background fields
+    if with_background
+        background_fields = Oceananigans.BackgroundFields(; 
+                             background_closure_fluxes=true, b=B̄_field)
+        # we want no flux bottom boundary (∂B∂z = 0) and infinite ocean at the top boundary
+        B_bcs = FieldBoundaryConditions(
+            bottom = GradientBoundaryCondition(-N^2), # ∂B∂z = 0 → ∂b∂z = -∂B∂z = -N²
+            top = GradientBoundaryCondition(0.) # ∂B∂z = ∂B̄∂z+∂b∂z = N² → ∂b∂z = 0 
+        );                                
+        model = NonhydrostaticModel(; grid, background_fields, tracers = :b, buoyancy=BuoyancyTracer(),
+                                boundary_conditions=(b = B_bcs,))
+        b = model.tracers.b
+        B̄ = model.background_fields.tracers.b
+        B = B̄ + b # total buoyancy field
+        filepath = "with_background.nc"
+    else
+        # again we want no flux bottom boundary (∂B∂z = 0) and infinite ocean at the top boundary
+        B_bcs = FieldBoundaryConditions(
+            bottom = GradientBoundaryCondition(0.), # ∂B∂z = 0
+            top = GradientBoundaryCondition(N^2) # ∂B∂z =  N²
+        );
+        model = NonhydrostaticModel(; grid, tracers = :b, buoyancy=BuoyancyTracer(),
+                                boundary_conditions=(b = B_bcs,))
+        Bᵢ(z) = constant_stratification(z, 0., (; N² = N^2))
+        set!(model, b=Bᵢ)  # add background buoyancy as an initial condition
+        b = model.tracers.b
+        B = b # total buoyancy field = perturbation buoyancy because there is no background buoyancy
+        filepath = "without_background.nc"
+    end
+    # remove output file if existed
+    !isfile(filepath) ? nothing : rm(filepath) 
+
+    # Run for a few iterations
+    simulation = Simulation(model, Δt=0.1, stop_iteration=5)
+    simulation.output_writers[:profile] = NetCDFOutputWriter(model, (B=B,),
+                                                schedule = IterationInterval(1),
+                                                verbose=true,
+                                                filename = filepath,
+                                                overwrite_existing = true)
```

Comment:
Is it necessary to use output here? I think you can directly return the value of `B` without output. That will make the test simpler and easier to maintain which is super important.